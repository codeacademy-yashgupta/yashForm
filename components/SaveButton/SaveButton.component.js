import React, { Component } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import { APP_THEME_COLOR, APP_TEXT_COLOR } from '../../constants';

export default class SaveButton extends Component {
  render() {
    return (
      <View>

        <TouchableOpacity
          onPress={this.props.onPress}
          style={{
            width: '100%', backgroundColor: APP_THEME_COLOR, position: 'absolute', bottom: 0, height: 50,
          }}
        >
          <Text style={{
            color: APP_TEXT_COLOR, fontSize: 22, textAlign: 'center', marginTop: 10,
          }}
          >
            SAVE

          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}
