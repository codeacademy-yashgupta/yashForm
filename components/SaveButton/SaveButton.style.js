import { APP_THEME_COLOR, APP_TEXT_COLOR } from '../../constants';

export default {
  buttonText: {
    color: APP_TEXT_COLOR, fontSize: 22, textAlign: 'center', marginTop: 10,
  },
  buttonView: {
    width: '100%', backgroundColor: APP_THEME_COLOR, position: 'relative', bottom: 0, height: 50,
  },
};
