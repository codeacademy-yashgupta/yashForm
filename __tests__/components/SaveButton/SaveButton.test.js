import React from 'react';
import renderer from 'react-test-renderer';
import SaveButton from '../../../components/SaveButton/SaveButton.component';
import 'react-native';

describe('SaveButton', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<SaveButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
