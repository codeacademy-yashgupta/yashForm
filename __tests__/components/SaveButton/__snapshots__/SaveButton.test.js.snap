// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButton should render correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#3700B3",
        "bottom": 0,
        "height": 50,
        "opacity": 1,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      SAVE
    </Text>
  </View>
</View>
`;
